contracts/protocols/mainnet/compound_iii/helpers.sol

```diff
@@ -0,0 +1,440 @@
+            _asset.liquidationFactor = asset.liquidationFactor;
+            _asset.supplyCap = asset.supplyCap;
+            _asset.totalCollateral = _comet.totalsCollateral(asset.asset).totalSupplyAsset;
+
+            assets[i] = _asset;
+        }
+    }
+
+    function getMarketConfig(address cometMarket) internal view returns (MarketConfig memory market) {
+        IComet _comet = IComet(cometMarket);
+        market.utilization = _comet.getUtilization();
+        market.assetCount = _comet.numAssets();
+        market.supplyRateInPercentWei = _comet.getSupplyRate(market.utilization);
+        market.borrowRateInPercentWei = _comet.getBorrowRate(market.utilization);
+        market.baseTrackingSupplySpeed = _comet.baseTrackingSupplySpeed();
+        market.baseTrackingBorrowSpeed = _comet.baseTrackingBorrowSpeed();
+        market.reservesInBase = _comet.getReserves();
+        market.storeFrontPriceFactor = _comet.storeFrontPriceFactor();
+        market.baseBorrowMinInBase = _comet.baseBorrowMin();
+        market.targetReservesInBase = _comet.targetReserves();
+        market.totalSupplyInBase = _comet.totalSupply();
+        market.totalBorrowInBase = _comet.totalBorrow();
+
+        market.baseToken = getBaseTokenInfo(_comet);
+        market.scales = getScales(_comet);
+
+        market.rewardConfig = new RewardsConfig[](1);
+        market.rewardConfig[0] = getRewardsConfig(cometMarket);
+        market.assets = getMarketAssets(_comet, market.assetCount);
+    }
+
+    function currentValue(
+        int104 principalValue,
+        uint64 baseSupplyIndex,
+        uint64 baseBorrowIndex,
+        uint64 baseIndexScale
+    ) internal view returns (int104) {
+        if (principalValue >= 0) {
+            return int104((uint104(principalValue) * baseSupplyIndex) / uint64(baseIndexScale));
+        } else {
+            return -int104((uint104(principalValue) * baseBorrowIndex) / uint64(baseIndexScale));
+        }
+    }
+
+    function isAssetIn(uint16 assetsIn, uint8 assetOffset) internal pure returns (bool) {
+        return (assetsIn & (uint16(1) << assetOffset) != 0);
+    }
+
+    function getBorrowableAmount(address account, address cometAddress) public returns (int256) {
+        IComet _comet = IComet(cometAddress);
+        UserBasic memory _userBasic = _comet.userBasic(account);
+        TotalsBasic memory _totalsBasic = _comet.totalsBasic();
+        uint8 _numAssets = _comet.numAssets();
+        address baseTokenPriceFeed = _comet.baseTokenPriceFeed();
+
+        int256 amount_ = int256(
+            (currentValue(
+                _userBasic.principal,
+                _totalsBasic.baseSupplyIndex,
+                _totalsBasic.baseBorrowIndex,
+                _comet.baseIndexScale()
+            ) * int256(_comet.getPrice(baseTokenPriceFeed))) / int256(1e8)
+        );
+
+        for (uint8 i = 0; i < _numAssets; i++) {
+            if (isAssetIn(_userBasic.assetsIn, i)) {
```

Comment:
@Richa-iitr what does this check?

Reply:
Check if the asset us a collateral for user's position